research/integrations/fedora-ci/index.md

```diff
@@ -0,0 +1,338 @@
+---
+title: Packit as Fedora dist-git CI
+authors: mfocko
+---
+
+Related links:
+
+- [Epic](https://github.com/packit/packit-service/issues/2453)
+- [Research issue](https://github.com/packit/packit-service/issues/2462)
+
+## Current Fedora CI
+
+1. Koji scratch build
+1. Installability
+1. tmt (if plans are present)
+
+### Zuul as a parallel CI
+
+1. Builts SRPM
+1. Submits Koji scratch build
+1. Runs RPM linter
+1. Runs RPM inspect
+1. Tests install
+1. Runs tmt
+
+## Possible implementation
+
+### SRPM build
+
+Ideally kept identical. Packit should not be involved in any way that can affect
+the results of the SRPM build, i.e., just run `fedpkg srpm` and be done with it.
+
+This can be done within the sandbox as we do for non-scratch builds. However it
+might need some tweaking as Packit has specific worfklow.
```

Comment:
This is a very interesting point -- do we want to be consistent as much as possible with Packit's upstream implementation or be close to the real Fedora builds.

I agree the suggested approach `fedpkg srpm` makes more sense from the user perspective since this results in the production-like result.

Reply:
How do we want to handle sources? (Do we want to continue with the current trend of requiring sources in the lookaside cache?)

Reply:
> This is a very interesting point -- do we want to be consistent as much as possible with Packit's upstream implementation or be close to the real Fedora builds.
> 
> I agree the suggested approach `fedpkg srpm` makes more sense from the user perspective since this results in the production-like result.

With respect to downstream, I would be definitely voting for avoiding adjusting the SRPM creation. Adjusting the SRPM creation in any way enables the chance of inconsistencies, i.e., I have green CI, but once I merge it doesn't build. I don't think many maintainers would like that :D

> How do we want to handle sources? (Do we want to continue with the current trend of requiring sources in the lookaside cache?)

I'm not really sure, but if we go deep into the possible improvements then it's quite possible that most of this initiative will not be achievable :)